probe-dictionary/explore.js

```diff
@@ -593,10 +603,37 @@ function updateSearchParams(pushState = false) {
     params.detailView = gDetailViewId;
   }
 
+  if (params['search']=="") {
+    delete params.search;
+  }
+  if (params['searchtype']=="in_any") {
+    delete params.searchtype;
+  }
+  if (params['optout']==false) {
+    delete params.optout;
+  }
+  if (params['channel']=="any") {
+    delete params.channel;
+  }
+  if (params['constraint']=="is_in") {
+    delete params.constraint;
+  }
+  if (params['version']=="any") {
+    delete params.version;
+  }
+  if (params['view']=="search-results-view") {
+    delete params.view;
+  }
+
+  let queryString = "";
+  if (Object.keys(params).length) {
+    queryString = "?" + $.param(params);
+  }
+  
   if (!pushState) {
-    window.history.replaceState("", "", "?" + $.param(params));
+    window.history.replaceState({}, "", queryString);
```

Comment:
Did you mean to replace the string with an object in these two cases?

Reply:
Actually, I looked for the [JavaScript Set/Update method](https://gist.github.com/excalq/2961415#file-gistfile1-js-L1). Both of the syntax work well. I am not very sure of the difference between the two. But this looked more appealing to me so I put that. Please guide which is more right. Thanks!